File: problems/002/002.jl
function solution()
  sum, curr, next = 0, 1, 2

  while curr < 4000000
    if curr % 2 == 0
      sum += curr
    end

    temp = next
    next += curr
    curr = temp
  end

  sum
end

if length(ARGS) == 1 && ARGS[1] == "-a"
  println(solution())
  exit(0)
end

for line in eachline(STDIN)
  iters = int(line)

  start = time_ns()
  for i in 1:iters
    solution()
  end
  end_ = time_ns()

  println(end_ - start)
end

File: problems/002/002.rb
def solution
  sum, curr, next_ = 0, 1, 2

  while curr < 4_000_000
    curr.even? && sum += curr

    temp = next_
    next_ += curr
    curr = temp
  end

  sum
end

if ARGV.length == 1 && ARGV[0] == '-a'
  puts solution
  exit
end

STDOUT.sync = true
STDIN.each do |line|
  iters = line.to_i

  start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  (1..iters).each do ||
    solution
  end
  end_ = Process.clock_gettime(Process::CLOCK_MONOTONIC)

  elapsed = ((end_ - start) * 1e9).to_i

  puts elapsed
end
